src/common/tests/CommonUtilsUT.cpp

```diff
@@ -1155,6 +1155,20 @@ TEST_F(CommonUtilsTest, FormatAllocateString)
     FREE_MEMORY(formattted);
 }
 
+TEST_F(CommonUtilsTest, FormatAllocateStringLong)
+{
+    int strLength = 4093;
+    char* formatted = nullptr;
+    std::string a(strLength, 'a');
+    std::string b(strLength, 'b');
+    formatted = FormatAllocateString("%s %s", a.c_str(), b.c_str());
+    EXPECT_NE(nullptr, formatted);
+    EXPECT_EQ(strLength * 2 + 1, strlen(formatted));
+    EXPECT_EQ(0, memcmp(formatted, a.c_str(), strLength));
+    EXPECT_EQ(0, memcmp(formatted + strLength + 1, b.c_str(), strLength));
+    FREE_MEMORY(formatted);
+}
+
```

Comment:
You may have noticed that although the wrapper of this test is C++, the actual test cases use C style for things such as allocating memory (with malloc), plain char* strings (not std::string), etc., in order to mimic real use of these functions under test as real as possible. 

It's not a big deal if not, but would you please consider:

1. Writing this test case in similar fashion.
2. Consider merging it with the existing FormatAllocateString test case.

We only need added one case for strings longer than 512 bytes, correct? (meaning, the test could be also simplified a bit)

Reply:
Ok, merged with `FormatAllocateString` and rewrote+simplified the test case.
One test case with a string longer than 512 bytes is enough.

Reply:
Awesome. Thank you!